aten/src/ATen/native/native_functions.yaml

```diff
@@ -1995,7 +1995,7 @@
   dispatch:
     CompositeExplicitAutograd: vdot_out
 
-- func: einsum(str equation, Tensor[] tensors) -> Tensor
+- func: einsum(str equation, Tensor[] tensors, *, int[]? path=None) -> Tensor
```

Comment:
@janeyx99 Looks like this needs an XLA-side patch (see XLA failures here:  https://app.circleci.com/pipelines/github/pytorch/xla/13558/workflows/a9174cd2-45f2-4426-ae5c-93b362837101/jobs/30725).

To unblock the failures: should we revert?

For the actual patch: I'm not familiar with opt_einsum, but my guess it can optimize on cpu and cuda, but not XLA? XLA has their own custom lowering for einsum (recently added [here](https://github.com/pytorch/xla/pull/3843)). If that's the case, we could just have XLA ignore the new path arg, or maybe error/warn if it was passed to avoid confusion. cc @JackCaoG.

Reply:
I just merged https://github.com/pytorch/xla/pull/4040. We need to coordinate landing next time.

Reply:
oh didn't see that :)